From: Felix Paul Kuehne <fkuehne@videolan.org>
Date: Wed, 12 Feb 2014 12:12:07 +0000 (+0100)
Subject: nporuntime: do not mix friend declarations with static implementations
X-Git-Url: http://git.videolan.org/?p=npapi-vlc.git;a=commitdiff_plain;h=f4a86b5994a797a36d94bb4cc8033efad3e7632b

nporuntime: do not mix friend declarations with static implementations

Fix compilation with clang++, part one
---

diff --git a/npapi/control/nporuntime.h b/npapi/control/nporuntime.h
index d67ec8c..b279ade 100644
--- a/npapi/control/nporuntime.h
+++ b/npapi/control/nporuntime.h
@@ -114,10 +114,10 @@ protected:
 
     friend void RuntimeNPClassDeallocate(NPObject *npobj);
     friend void RuntimeNPClassInvalidate(NPObject *npobj);
-    template <class RuntimeNPObject> friend bool RuntimeNPClassGetProperty(NPObject *npobj, NPIdentifier name, NPVariant *result);
-    template <class RuntimeNPObject> friend bool RuntimeNPClassSetProperty(NPObject *npobj, NPIdentifier name, const NPVariant *value);
-    template <class RuntimeNPObject> friend bool RuntimeNPClassRemoveProperty(NPObject *npobj, NPIdentifier name);
-    template <class RuntimeNPObject> friend bool RuntimeNPClassInvoke(NPObject *npobj, NPIdentifier name,
+    template <class RuntimeNPObject> static bool RuntimeNPClassGetProperty(NPObject *npobj, NPIdentifier name, NPVariant *result);
+    template <class RuntimeNPObject> static bool RuntimeNPClassSetProperty(NPObject *npobj, NPIdentifier name, const NPVariant *value);
+    template <class RuntimeNPObject> static bool RuntimeNPClassRemoveProperty(NPObject *npobj, NPIdentifier name);
+    template <class RuntimeNPObject> static bool RuntimeNPClassInvoke(NPObject *npobj, NPIdentifier name,
                                                     const NPVariant *args, uint32_t argCount,
                                                     NPVariant *result);
     friend bool RuntimeNPClassInvokeDefault(NPObject *npobj,
